token.go

```diff
@@ -285,7 +285,7 @@ func (c *Connection) sendTokenFormTimed(ctx context.Context, form url.Values) (c
 
 	// Check the response status and content type:
 	code = response.StatusCode
-	if response.StatusCode != http.StatusOK {
+	if response.StatusCode >= 500 {
```

Comment:
I think it would be better to preserve the check for StatusOK and move it to line 314; right after checking if the JSON response contains an error message. Otherwise you are assuming that 5XX responses don't have a JSON body, which may not be true.